pkg/odo/cli/component/create.go

```diff
@@ -1079,8 +1079,8 @@ func NewCmdCreate(name, fullName string) *cobra.Command {
 	componentCreateCmd.Flags().StringSliceVar(&co.componentEnvVars, "env", []string{}, "Environmental variables for the component. For example --env VariableName=Value")
 
 	if experimental.IsExperimentalModeEnabled() {
-		componentCreateCmd.Flags().StringVar(&co.devfileMetadata.downloadSource, "downloadSource", "", "Download sample project from devfile.")
-		componentCreateCmd.Flags().Lookup("downloadSource").NoOptDefVal = defaultProjectName //Default value to pass to the flag if one is not specified.
+		componentCreateCmd.Flags().StringVar(&co.devfileMetadata.starter, "starter", "", "Download starter project from devfile.")
```

Comment:
Honestly, so weird using this wording. Because if you try to do `odo create` with just: `--starter` and `--devfile`, example:

```
odo create --starter --devfile https://github.com/odo-devfiles/nodejs-ex/archive/0.0.1.zip
```

It makes it seem like it's downloading a "sample" project, but it's not, it's downloading the actual project specified in `projects` under the `devfile.yaml` file, if that makes sense?

Reply:
@cdrage There's a separate story open to use the new `starterProject` field in devfile v2 https://github.com/openshift/odo/issues/3263 with `--starter`, but `starterProject` hasn't been finalized in the schema yet.

Reply:
@cdrage I agree with you this needs to be fixed. But that is a separate issue. 
This is just about renaming this flag. It is supposed to download the sample (starter) project from the devfile the original name was creating a lot of confusion. Once the `startProject` field is added to devfile spec we can fix the rest.